naming/src/main/java/com/alibaba/nacos/naming/push/PushService.java

```diff
@@ -122,64 +123,66 @@ public void onApplicationEvent(ServiceChangeEvent event) {
         String serviceName = service.getName();
         String namespaceId = service.getNamespaceId();
         
-        Future future = GlobalExecutor.scheduleUdpSender(() -> {
```

Comment:
The future is useful, can't be removed.

Reply:
<img width="429" alt="future" src="https://user-images.githubusercontent.com/40652892/128444579-62ef3139-7648-4e78-8e22-7a299a11e551.png">
why?

Reply:
I mean, the future is been used to merge change event. No need to do this change.

Reply:
servcie change event merge alogrithm is changed to ConcurrentMap<String, Long> that has nothing to do with future.
please take a full look at all changes.

Reply:
we have more than 100 instances per service,  all the instances will be started or  restarted by one touch click on our control system, the sevice merge mechanism using Future will be broken: all the futureMap.contains() calling almost return false because the future is not been put into the futureMap.
we brought the merge mechanism forward to publish event stage by doing theses changes, the push frequency was reduced eventually.
@KomachiSion

Reply:
the target of futureMap.contains() is to merge the frequently changes for same service. the future will be create even you ignore the `return`. So Why don't you use it but use a new `Long`? I has no idea the benefits of your changes.

Reply:
`onApplicationEvent` is a single thread I remember. spring event system will keep the event orders.

Reply:
serviceChanged() is not running at single thread:
Service.onChange -> updateIPs ->> PushService.serviceChanged is not running  at single thread mode, if more than one instances of the same service start up at the same time, more than one serviceChangeEvent of the same service might be created and published to ApplicationContext, all the events will be processed by onApplicationEvent()  successively.

Reply:
I think there is no problem, If you think it will cause your problem, should we double check futureMap in `onApplicationEvent` rather than do much in `serviceChanged`?